src/common/survival_util.cc

```diff
@@ -0,0 +1,255 @@
+#include <dmlc/registry.h>
+#include <cstdlib>
+#include <xgboost/enum_class_param.h>
+#include "survival_util.h"
+
+namespace xgboost {
+namespace common {
+
+DMLC_REGISTER_PARAMETER(AFTParam);
+
+AFTDistribution* AFTDistribution::Create(AFTDistributionType dist) {
+  switch (dist) {
+   case AFTDistributionType::kNormal:
+    return new AFTNormal;
+   case AFTDistributionType::kLogistic:
+    return new AFTLogistic;
+   case AFTDistributionType::kExtreme:
+    return new AFTExtreme;
+   default:
+    LOG(FATAL) << "Unknown distribution";
+  }
+  return nullptr;
+}
+
+double AFTNormal::pdf(double z) {
+  double pdf;
+  pdf = (std::exp(-std::pow(z/std::sqrt(2),2)))/std::sqrt(2*kPI);
```

Comment:
@avinashbarnwal no need to use std::sqrt(2) inside, take it out and avoid a call to sqrt. also I would remove std::pow, e.g.
```
  pdf = std::exp(-z*z/2)/std::sqrt(2*kPI);
```
that is easier to read, and should be faster to evaluate